### DISCUSSION TITLE
very high RAM usage in docker container (wsl2)

### DISCUSSION THREAD

**Discussion by valiant8086** [AzuraCast/AzuraCast#6195] [Q&A]
i3 12100 desktop with 16gb of RAM with Windows 10. We run a Windows10 vm that has 4gb of RAM under VMware on this system. We just installed and are learning Azuracast, but it has very high RAM usage about 6gb. On the dashboard, it says it is using 8gb, but vmvm.exe is reading around 6gb. Our machine hovers around 90% RAM usage. We would like to find out how to control this. the VMware vm is actually using a lot less to run Windows 10 than this docker container needs just to run Ubuntu that has no gui with it or anything like that, just Azuracast is the only thing installed on it. I really thought we could get by with less than 4gb no matter what given that it's just Linux, but it promptly goes to 5 as soon as it launches and shamelessly creeps up from there. It doesn't seem to be climbing continuously, so perhaps not a memory leak.

I can run Ubuntu on 4gb of memory in VMware, but I like the accessibility of wsl2 if we can learn how to curb this.

**Comment by BusterNeece** [comment]
Personally, I specifically limit the amount of CPU and RAM that WSL2 can use on my system via `.wslconfig` files: https://learn.microsoft.com/en-us/windows/wsl/wsl-config

This helps give me plenty of extra system resources for other things and ensures the Linux side never dominates things so much that it hangs my entire system or impacts my IDE.
